mlflow/server/js/src/components/artifact-view-components/ShowArtifactMapView.js

```diff
@@ -0,0 +1,132 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { getSrc } from './ShowArtifactPage';
+import './ShowArtifactMapView.css';
+import { getRequestHeaders } from '../../setupAjaxHeaders';
+import 'leaflet/dist/leaflet.css';
+import L from 'leaflet';
+import icon from 'leaflet/dist/images/marker-icon.png';
+import iconRetina from 'leaflet/dist/images/marker-icon-2x.png';
+import iconShadow from 'leaflet/dist/images/marker-shadow.png';
+
+function onEachFeature(feature, layer) {
+  if (feature.properties && feature.properties.popupContent) {
+    const popupContent = feature.properties.popupContent;
+    layer.bindPopup(popupContent);
+  }
+}
+
+class ShowArtifactMapView extends Component {
+  constructor(props) {
+    super(props);
+    this.fetchArtifacts = this.fetchArtifacts.bind(this);
+    this.leafletMap = undefined;
+  }
+
+  static propTypes = {
+    runUuid: PropTypes.string.isRequired,
+    path: PropTypes.string.isRequired,
+  };
+
+  state = {
+    loading: true,
+    error: undefined,
+    features: undefined,
+  };
+
+  componentDidMount() {
+    this.fetchArtifacts();
+  }
+
+  componentDidUpdate(prevProps) {
+    if (this.props.path !== prevProps.path || this.props.runUuid !== prevProps.runUuid) {
+      this.fetchArtifacts();
+    }
+
+    if (this.leafletMap !== undefined) {
+      if (this.leafletMap.hasOwnProperty('_layers')) {
+        this.leafletMap.off();
+        this.leafletMap.remove();
+        document.getElementsByClassName('map-container')[0].innerHTML = "<div id='map'></div>";
```

Comment:
Nit: could we pull out `'map'` into a constant? It would help make it clearer that the ID of this div is what we're passing into the leaflet APIs further below (e.g. when we call `L.map('map')` below)

Reply:
Done